src/snowflake/snowpark/_internal/compiler/cte_utils.py

```diff
@@ -45,9 +45,40 @@ def find_duplicate_subtrees(
 
     This function is used to only include nodes that should be converted to CTEs.
     """
-    id_count_map = defaultdict(int)
+    id_node_map = defaultdict(list)
     id_parents_map = defaultdict(set)
-    id_complexity_map = defaultdict(int)
```

Comment:
I don't see the benefit of this approach over the previous approach. In this change, we are appending nodes to id_node_map list where we only access list[0] to calculate the complexity score.

Reply:
This is not needed anymore with the current change, the benefit is to avoid unnecessary data  structure, and it also cleaned up some unnecessary code  during the tree travel.

Reply:
I think appending nodes to a list only to get len of list later is not very efficient but I don't strongly disagree with your changes if it allows cleanup some code.

Reply:
Oh, the main reason I need node here is because i need the node to help me getting the node property later, not really for clean up purpose, the clean up is a side effect after updating to the new data structure.